### PR TITLE
fix(trainer): step LR scheduler only on actual optimizer updates

### LINKED ISSUE #204: Unexpected LR Increase with cosine_with_min_lr Scheduler
Using cosine_with_min_lr, the learning rate unexpectedly increases during training (e.g., steps 30k-40k) after warmup. I save checkpoints every 10,000 steps and resume training from them.

My config:
```
Framework_name=QwenOFT
freeze_module_list=''
base_vlm=playground/Pretrained_models/Qwen2.5-VL-3B-Instruct
config_yaml=./examples/LIBERO/train_files/starvla_cotrain_libero.yaml
libero_data_root=playground/Datasets/LEROBOT_LIBERO_DATA
data_mix=libero_all
run_root_dir=./results/Checkpoints
run_id=0311_libero4in1_qwen25oft
#######################################################
output_dir=${run_root_dir}/${run_id}
mkdir -p ${output_dir}
cp $0 ${output_dir}/

accelerate launch \
  --config_file starVLA/config/deepseeds/deepspeed_zero2.yaml \
  --num_processes 8 \
  starVLA/training/train_starvla.py \
  --config_yaml ${config_yaml} \
  --framework.name ${Framework_name} \
  --framework.qwenvl.base_vlm ${base_vlm} \
  --datasets.vla_data.data_root_dir ${libero_data_root} \
  --datasets.vla_data.data_mix ${data_mix} \
  --datasets.vla_data.per_device_batch_size 8 \
  --trainer.vla_data.video_backend torchvision_av \
  --trainer.freeze_modules "${freeze_module_list}" \
  --trainer.max_train_steps 80000 \
  --trainer.save_interval 10000 \
  --trainer.logging_frequency 100 \
  --trainer.eval_interval 100 \
  --trainer.gradient_accumulation_steps 2 \
  --trainer.is_resume true \
  --trainer.resume_step 30000 \
  --run_root_dir ${run_root_dir} \
  --run_id ${run_id} \
  --wandb_project starVLA_Libero_qwen25_OFT_30000 \
  --wandb_entity xxx \
  --is_debug False
```
Could this be related to resume logic?
Image

### PR DESCRIPTION
**Pull request by wadeKeith**
## Problem

When using gradient_accumulation_steps > 1, lr_scheduler.step() is called on every micro-batch inside accelerator.accumulate(), but optimizer.step() is a no-op on non-sync micro-steps. This causes the LR scheduler to advance gradient_accumulation_steps times faster than intended.

### Impact

1. **Normal training**: LR decays faster than configured — with gradient_accumulation_steps=2, the effective num_training_steps for the scheduler is halved.

2. **Resume training (#204)**: _adjust_lr_scheduler_for_resume() replays completed_steps (which counts optimizer steps), but the scheduler internal counter was advancing at the micro-batch rate. On resume, the scheduler restarts from 0 and is stepped completed_steps times, but this does not match the phase it was at before the interruption. The result is an unexpected LR spike after resuming.

## Fix

Guard lr_scheduler.step() with self.accelerator.sync_gradients so it only advances when an actual optimizer update occurs.

Applied to both train_starvla.py and train_starvla_cotrain.py.

Closes #204

### CODE REVIEW

**Comment by wadeKeith** [comment]
Friendly ping — just checking in on this PR to see if you have any feedback. Happy to address any questions! 🙏

**Review by JinhuiYE — APPROVED** [review]
Hi, You're right, the update looks good. I'll merge it now.
